openmp/tools/omptest/include/OmptAssertEvent.h

```diff
@@ -0,0 +1,377 @@
+//===- OmptAssertEvent.h - Assertion event declarations ---------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// Contains assertion event constructors, for generally all observable events.
+/// This includes user-generated events, like synchronization.
+///
+//===----------------------------------------------------------------------===//
+
+#ifndef OPENMP_TOOLS_OMPTEST_INCLUDE_OMPTASSERTEVENT_H
+#define OPENMP_TOOLS_OMPTEST_INCLUDE_OMPTASSERTEVENT_H
+
+#include "InternalEvent.h"
+#include "omp-tools.h"
+
+#include <cassert>
+#include <limits>
+#include <memory>
+#include <string>
+
+namespace omptest {
+
+enum class ObserveState { generated, always, never };
```

Comment:
This uses the third spelling scheme for values inside an `enum class`. 
I think we should also capitalize the spelling to follow what we do in `enum class EventTy`